bin/report

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# bin/report <build-dir> <cache-dir> <env-dir>
+
+### Configure environment
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+BUILD_DIR=${1:-}
+
+# <separator> <array>
+joinArrayWithSeparator() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+packages=()
+custom_packages=()
+custom_repositories=()
+
+while IFS= read -r line; do
+  if [[ $line ]]; then
```

Comment:
This matches on any non-empty line, so you'll want to check for lines that only contain whitespace.

You can however instead add `-e '^$'` to the `grep` further down, and then get rid of this wrapping `if` entirely.

Doing `-e '^\s*$'` instead will also drop lines that contain only whitespace.

Reply:
i'm replacing this initial conditional to `if grep --silent --invert-match -e "^#" -e "^\s*$" <<< "${line}"; then` so that blank lines and comments are ignored.